app/upload.go

```diff
@@ -24,10 +24,19 @@ const minFirstPartSize = 5 * 1024 * 1024 // 5MB
 
 func (a *App) genFileInfoFromReader(name string, file io.ReadSeeker, size int64) (*model.FileInfo, error) {
 	ext := strings.ToLower(filepath.Ext(name))
+
 	info := &model.FileInfo{
-		Name:     name,
-		MimeType: mime.TypeByExtension(ext),
+		Name:      name,
+		MimeType:  mime.TypeByExtension(ext),
+		Size:      size,
+		Extension: ext,
+	}
+
+	if ext != "" && ext[0] == '.' {
```

Comment:
Is the `ext[0]` check necessary? The API https://pkg.go.dev/path/filepath#Ext guarantees that if it's non-empty, the dot will always be there.

Reply:
It doesn't look like it's needed, let me update.